tcf_website/static/reviews/review_form.js

```diff
@@ -0,0 +1,136 @@
+        $("#courseID").prop("disabled", false);
+        $("#instructor").prop("disabled", true);
+        $("#semester").prop("disabled", true);
+
+        // Fetch course data from API, based on selected subdepartment
+        var subdeptID = $("#subject").val();
+        var pageSize = "1000";
+        var courseEndpoint = "/api/courses/?subdepartment=" + subdeptID +
+                             "&page_size=" + pageSize + "&recent";
+        $.getJSON(courseEndpoint, function(data) {
+            // Generate option tags
+            $.each(data.results, function(i, course) {
+                $("<option />", {
+                    val: course.id,
+                    text: course.number
+                }).appendTo("#courseID");
+            });
+            return this;
+        });
+    });
+
+    // Fetch instructor data on course select
+    $("#courseID").change(function() {
+        // Clear & disable sequenced dropdowns
+        clearDropdown("#instructor");
+        clearDropdown("#semester");
+        $("#courseName").val("");
+        // Enable instructor selector, disable the following
+        $("#instructor").prop("disabled", false);
+        $("#semester").prop("disabled", true);
+
+        // Update course name field
+        var courseID = $("#courseID").val();
+        var courseEndpoint = "/api/courses/" + courseID;
+        $.getJSON(courseEndpoint, function(course) {
+            $("#courseName").val(course.title);
+        });
+
+        // Fetch instructor data from API, based on selected course
+        var pageSize = "1000";
+        var instrEndpoint = "/api/instructors/?section__course=" + courseID +
+                            "&page_size=" + pageSize + "&recent=true";
+        $.getJSON(instrEndpoint, function(data) {
+            clearDropdown("#instructor"); // Empty dropdown
+
+            // Generate option tags
+            $.each(data.results, function(i, instr) {
+                $("<option />", {
+                    val: instr.id,
+                    text: instr.first_name + " " + instr.last_name
+                }).appendTo("#instructor");
+            });
+            return this;
+        });
+    });
+
+    // Fetch semester data on instructor select
+    $("#instructor").change(function() {
+        // Clear & disable sequenced dropdowns
+        clearDropdown("#semester");
+        // Enable semester selector, disable the following
+        $("#semester").prop("disabled", false);
+
+        // Fetch all semester data from API
+        var courseID = $("#courseID").val();
+        var instrID = $("#instructor").val();
+        var semEndpoint = "/api/semesters/?course=" + courseID +
+                          "&instructor=" + instrID;
+        $.getJSON(semEndpoint, function(data) {
+            // Generate option tags
+            $.each(data, function(i, semester) {
+                // Note: API returns semester list in reverse chronological order
+                // Most recent 5 years only
```

Comment:
Do we still need this `if` statement?

Reply:
probably not hmm

Reply:
Should I add it in the backend? I think it's a good practice to minimize business logic in the frontend.

Reply:
Sure, go for it! feel free to just make the commit on this PR

Reply:
Done in 2d3e6890091505a617c3dc6054ee13c06a561f10

Reply:
Can you change accordingly in the frontend?

Reply:
@Bartleby2718 Does this only return most recent now? It's not a param?

Reply:
Seems to be the case based on `InstructorFilter.filter_recent_instructors_by_course` in `/api/filters.py` ; it doesn't do a boolean check anymore but I'm not sure what `value` in the function does now.

Reply:
Alex is correct about this, and I made the change because of #259. As I said in another thread, `value` in the function is the value of the `course` parameter.

Reply:
Gotcha. Does this affect what's shown on the instructor pages?

Reply:
If you're only going to show recent courses then the instructor averages should probably update accordingly
^I think showing only recent ratings is prob a better representation of the instructor's current proficiency (ex. Venkat)

Reply:
Oh wait are we even using the API for instructor pages?

Reply:
> Gotcha. Does this affect what's shown on the instructor pages?

Nope, because it doesn't use the API.

> If you're only going to show recent courses then the instructor averages should probably update accordingly

^I think showing only recent ratings is prob a better representation of the instructor's current proficiency (ex. Venkat)
We should definitely discuss our definition of recency and where we'll use it, but it won't be too difficult, although complicated.

> Oh wait are we even using the API for instructor pages?

Nope, because performance is not an issue there.